automata/fa/dfa.py

```diff
@@ -1476,22 +1753,25 @@ def compress(word: str, next_word: str) -> None:
 
         compress(prev_word, "")
 
-        # Add dump state. Always needed since dict is finite
-        dump_state = 0
-        transitions[dump_state] = {symbol: dump_state for symbol in input_symbols}
+        if not as_partial_dfa:
+            # Add trap state. Always needed since dict is finite
+            trap_state = 0
+            transitions[trap_state] = {symbol: trap_state for symbol in input_symbols}
 
-        for path in transitions.values():
-            for symbol in input_symbols:
-                path.setdefault(symbol, dump_state)
+            for path in transitions.values():
+                for symbol in input_symbols:
+                    path.setdefault(symbol, trap_state)
 
         return cls(
             states=frozenset(transitions.keys()),
             input_symbols=input_symbols,
             transitions=transitions,
             initial_state="",
             final_states=final_states,
+            allow_partial=as_partial_dfa,
         )
 
+    # Supports partial
     @classmethod
     def from_nfa(
```

Comment:
@EduardoGoulart1 This function should have an `as_partial` option. This is because we always know that the state corresponding to the empty set is a trap state, so if `as_partial` is set, we can always omit and avoid traversing the empty set in the BFS. This should substantially speed up this algorithm when used for converting NFAs that come from regexs (since these NFAs have a lot of transitions to the empty set for unused characters).

Reply:
Ok will do... I am of the more radical opinion that there should only be partial DFAs

Reply:
This should be pretty straightforward, as I added some of this functionality in my commits.

Also, I actually think that's not such a radical idea. We can discuss semantics when more things are implemented, but there might be a nice way of making the partial property appear in more places.